Content.Shared/Ninja/Components/SpaceNinjaGlovesComponent.cs

```diff
@@ -0,0 +1,134 @@
+using Content.Shared.Actions;
+using Content.Shared.Actions.ActionTypes;
+using Content.Shared.Ninja.Systems;
+using Content.Shared.Tag;
+using Robust.Shared.Audio;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+using Robust.Shared.Utility;
+using System.Threading;
+
+namespace Content.Shared.Ninja.Components;
+
+/// <summary>
+/// Component for toggling glove powers.
+/// Powers being enabled is controlled by GlovesEnabledComponent
+/// </summary>
+[Access(typeof(SharedNinjaSystem))]
+[RegisterComponent]
+public sealed class SpaceNinjaGlovesComponent : Component
+{
+    /// <summary>
+    /// The action for toggling ninja gloves abilities
+    /// </summary>
+    [DataField("toggleAction")]
+    public InstantAction ToggleAction = new()
+    {
+        DisplayName = "action-name-toggle-ninja-gloves",
+        Description = "action-desc-toggle-ninja-gloves",
+        Priority = -13,
+        Event = new ToggleNinjaGlovesEvent()
+    };
+}
+
+/// <summary>
+/// This marker component enables Space Ninja Gloves' abilities, usable on clicking something using an empty hand.
+/// </summary>
+[Access(typeof(SharedNinjaSystem))]
+[RegisterComponent]
+public sealed class GlovesEnabledComponent : Component
+{
+}
```

Comment:
Looks like this is just here to show that gloves are enabled? You should be able to do that with a bool on the gloves instead.

Reply:
thats what i did originally, but it's simpler to HasComp<GlovesEnabledComponent>(uid) than TryComp<GlovesComponent>(uid, out var gloves) && gloves.Enabledp
lus its inline with the EmaggedComponent kinda thing with a component instead of a bool on random thing

Reply:
It's still redundant to have this in favor of having a bool. You shouldn't have to do a trycomp for gloves on the even when you swap it to AfterInteract.

Reply:
gone back to bool now